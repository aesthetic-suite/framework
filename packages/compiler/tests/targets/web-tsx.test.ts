import { runTargetTests } from './runTargetTests';

runTargetTests('web', 'web-tsx', true);
