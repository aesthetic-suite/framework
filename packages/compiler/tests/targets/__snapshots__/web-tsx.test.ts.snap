// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format web-tsx compiles and writes files 1`] = `
"/* Auto-generated by Aesthetic. Do not modify! */
/* eslint-disable */

import { LocalBlock } from '@aesthetic/sss';
import { Design } from '@aesthetic/system';

export default new Design<LocalBlock>('test', {
  border: {
    sm: {
      radius: '0.11rem', // 1.50px
      width: '0rem', // 0px
    },
    df: {
      radius: '0.21rem', // 3px
      width: '0.07rem', // 1px
    },
    lg: {
      radius: '0.32rem', // 4.50px
      width: '0.14rem', // 2px
    },
  },
  breakpoint: {
    xs: {
      query: 'screen and (min-width: 45.71em)',
      querySize: 640,
      rootLineHeight: 1.33,
      rootTextSize: '14.94px',
    },
    sm: {
      query: 'screen and (min-width: 68.57em)',
      querySize: 960,
      rootLineHeight: 1.42,
      rootTextSize: '15.94px',
    },
    md: {
      query: 'screen and (min-width: 91.43em)',
      querySize: 1280,
      rootLineHeight: 1.52,
      rootTextSize: '17.01px',
    },
    lg: {
      query: 'screen and (min-width: 114.29em)',
      querySize: 1600,
      rootLineHeight: 1.62,
      rootTextSize: '18.15px',
    },
    xl: {
      query: 'screen and (min-width: 137.14em)',
      querySize: 1920,
      rootLineHeight: 1.73,
      rootTextSize: '19.36px',
    },
  },
  heading: {
    l1: {
      letterSpacing: '0.25px',
      lineHeight: 1.5,
      size: '1.14rem', // 16px
    },
    l2: {
      letterSpacing: '0.33px',
      lineHeight: 1.69,
      size: '1.43rem', // 20px
    },
    l3: {
      letterSpacing: '0.44px',
      lineHeight: 1.9,
      size: '1.79rem', // 25px
    },
    l4: {
      letterSpacing: '0.59px',
      lineHeight: 2.14,
      size: '2.23rem', // 31.25px
    },
    l5: {
      letterSpacing: '0.79px',
      lineHeight: 2.4,
      size: '2.79rem', // 39px
    },
    l6: {
      letterSpacing: '1.05px',
      lineHeight: 2.7,
      size: '3.49rem', // 48.80px
    },
  },
  shadow: {
    xs: {
      x: '0rem', // 0px
      y: '0.14rem', // 2px
      blur: '0.14rem', // 2px
      spread: '0rem', // 0px
    },
    sm: {
      x: '0rem', // 0px
      y: '0.23rem', // 3.24px
      blur: '0.27rem', // 3.75px
      spread: '0rem', // 0px
    },
    md: {
      x: '0rem', // 0px
      y: '0.37rem', // 5.24px
      blur: '0.39rem', // 5.50px
      spread: '0rem', // 0px
    },
    lg: {
      x: '0rem', // 0px
      y: '0.61rem', // 8.47px
      blur: '0.52rem', // 7.25px
      spread: '0rem', // 0px
    },
    xl: {
      x: '0rem', // 0px
      y: '0.98rem', // 13.71px
      blur: '0.64rem', // 9px
      spread: '0rem', // 0px
    },
  },
  spacing: {
    xs: '0.31rem', // 4.38px
    sm: '0.63rem', // 8.75px
    df: '1.25rem', // 17.50px
    md: '2.50rem', // 35px
    lg: '3.75rem', // 52.50px
    xl: '5rem', // 70px
    type: 'vertical-rhythm',
    unit: 17.5,
  },
  text: {
    sm: {
      lineHeight: 1.25,
      size: '0.89rem', // 12.45px
    },
    df: {
      lineHeight: 1.25,
      size: '1rem', // 14px
    },
    lg: {
      lineHeight: 1.25,
      size: '1.13rem', // 15.75px
    },
  },
  typography: {
    font: {
      heading:
        'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"',
      locale: {},
      monospace:
        'Menlo, Monaco, Consolas, \\"Lucida Console\\", \\"Liberation Mono\\", \\"Courier New\\", monospace',
      text:
        'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"',
      system:
        'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"',
    },
    rootLineHeight: 1.25,
    rootTextSize: '14px',
  },
});
"
`;

exports[`Format web-tsx compiles and writes files 2`] = `
"/* Auto-generated by Aesthetic. Do not modify! */
/* eslint-disable */

import design from '..';

export default design.createTheme(
  {
    contrast: 'normal',
    scheme: 'light',
  },
  {
    palette: {
      brand: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      primary: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      secondary: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      tertiary: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      neutral: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      muted: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      danger: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      warning: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      negative: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
      positive: {
        color: {
          '00': '#000',
          '10': '#000',
          '20': '#000',
          '30': '#000',
          '40': '#000',
          '50': '#000',
          '60': '#000',
          '70': '#000',
          '80': '#000',
          '90': '#000',
        },
        bg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        fg: {
          base: '#000',
          focused: '#000',
          hovered: '#000',
          selected: '#000',
          disabled: '#000',
        },
        text: '#000',
      },
    },
  },
);
"
`;
